dataplane/src/pipeline/dyn_nf.rs

```diff
@@ -28,6 +28,13 @@ pub trait DynNetworkFunction<Buf: PacketBufferMut> {
     /// [`DynPipeline::process`] with a concrete iterator type.  However, if you only
     /// have a dynamic iterator, you can use this method to process the packets.
     fn process_dyn<'a>(&'a mut self, input: DynIter<'a, Packet<Buf>>) -> DynIter<'a, Packet<Buf>>;
+
+    /// Allow objects implementing [`DynNetworkFunction`] to identify themselves by overriding this.
+    #[allow(dead_code)]
+    #[allow(clippy::unnecessary_literal_bound)]
+    fn nf_name(&self) -> &str {
+        "anonymous-dynNF"
```

Comment:
Is there a rust introspection feature that would allow us to calculate the type name of Self to print out?

I believe this is a good idea in concept, but I want to think about the api ergonomics a bit before approving.

Reply:
Doing some homework, it may be best to just add this to the trait and then punt on the macros/introspection as it seems pretty complex and not worth it right now.  Especially because we only want a partial implementation of the trait.  @daniel-noland can you chime in on the effort required for something like 
```
#[derive(NetworkFunction)]
struct MyStage;
```

Where this would add the required trait and will use name as `MyStage`.  But later the user should be able to do
```
impl NetworkFunction for MyStage {
  fn process(...)
}
```

Reply:
Let's leave it as is, fix it later.